packages/frontend/src/components/wallet-migration/WalletMigration.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+
+import { selectAvailableAccounts } from '../../redux/slices/availableAccounts';
+import {
+    keyToString,
+    generatePublicKey,
+    encodeAccountsTo
+} from '../../utils/encoding';
+import { getMyNearWalletUrlFromNEARORG } from '../../utils/getWalletURL';
+import { getLedgerHDPath } from '../../utils/localStorage';
+import { wallet } from '../../utils/wallet';
+import MigrateAccounts from './MigrateAccounts';
+import MigrationPrompt from './MigrationPrompt';
+import MigrationSecret from './MigrationSecret';
+import SelectDestinationWallet from './SelectDestinationWallet';
+
+
+export const WALLET_MIGRATION_VIEWS = {
+    MIGRATION_PROMPT: 'MIGRATION_PROMPT',
+    MIGRATION_SECRET: 'MIGRATION_SECRET',
+    SELECT_DESTINATION_WALLET: 'SELECT_DESTINATION_WALLET',
+    MIGRATE_ACCOUNTS: 'MIGRATE_ACCOUNTS'
+};
+
+const initialState = {
+    activeView: null,
+    walletType: null,
+    migrationKey: generatePublicKey()
+};
+
+
+
+const encodeAccountsToURL = async (accounts, publicKey) => {
+    const accountsData = [];
+    for (let i = 0; i < accounts.length; i++) {
+        const accountId = accounts[i];
+        const keyPair = await wallet.getLocalKeyPair(accountId);
+        accountsData.push([
+            accountId,
+            keyPair.secretKey,
```

Comment:
I get an error when I try to migrate accounts:
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/16845555/176855362-c2d956e6-5a98-4217-b724-f75d3d8ff460.png">